src/core/core.datasetController.js

```diff
@@ -370,6 +370,11 @@ export default class DatasetController {
     const dataset = me.getDataset();
     let stackChanged = false;
 
+    let labels = me.chart.data.labels;
```

Comment:
This would not work if the labels are defined on scale or xLabels / yLabels are used.

You should also report the bug this is supposed to fix, I'd tend to think the fix should be in category scale.

Reply:
Took a better look and I was probably wrong about the fix belonging to category scale.
I thing it would be bettor to use `meta.iScale.getLabels()` to get the right labels array.